src/JobPanel/ui/menus/main_menu_bar.py

```diff
@@ -124,6 +124,41 @@ def lock_by_status(self, rdeleted, downloaded, mj_local, task_status=None):
             else:
                 menu_action.setDisabled(False)
 
+    def lock_for_selection(self, multijobs, mj_ids):
+        """
+        Locks UI actions based on selected MultiJobs.
+        Disables those actions in UI, which don't make sense.
+        :param multijobs: list of all multijobs
+        :param mj_ids: list ids of selected multijobs
+        """
+        self.actionDownloadWholeMultiJob.setDisabled(True)
+        self.actionReuseMultiJob.setDisabled(True)
+        self.actionSendReport.setDisabled(True)
+
+        disable_remote_delete = True
+        for mj_id in mj_ids:
+            if ((multijobs[mj_id].state.status, MultijobActions.delete_remote) in TASK_STATUS_PERMITTED_ACTIONS) and not multijobs[mj_id].preset.deleted_remote:
```

Comment:
Modify MultiJobActions to instances of a new class MJAction with a method:
MJAction.permitted_mj_states().

Rewrite as:
mj_selected_states = [ multijobs[id].state.status for id in mj_ids ]
disable_remote_delete = any([ state in MultijobActions.delete_remote.permitted_states for st in mj_selected_states])